app/furniture/marketplace/cart/delivery_area_component.html.erb

```diff
@@ -4,18 +4,20 @@
     <%- if delivery_area.present? %>
       <div>
         <%= delivery_area.label %>
-
         <div class="text-sm text-gray-500">
           <%= render(Marketplace::Cart::DeliveryExpectationsComponent.new(cart: cart)) %>
         </div>
-
       </div>
-
       <div class="text-right">
         <%= link_to(t('marketplace.cart.delivery_areas.edit.link_to'), cart.location(:edit, child: :delivery_area)) %>
       </div>
     <%- else %>
-      <%= render "marketplace/cart/delivery_areas/form", cart: cart %>
+      <%- if cart.marketplace.has_single_delivery_area? %>
```

Comment:
I'm curious about why you chose to define `has_single_delivery_area?` (note: Ruby prefers `single_delivery_area?` over `has_...` on the `marketplace` while the `single_delivery_area_label` lives on the component?

Since both are really only used here, I tend to keep my method definitions as close to their sole caller as possible, and then "promote" methods that want to be used elsewhere.

Reply:
Good suggestion, I agree with putting this method on the component. I just didn't think through it very hard before comitting the code because I implemented the _idea_ on `marketplace` before I knew where I was putting it in the UI code. At the time `marketplace` felt "closer" in my brain. :)